elastic/kibana
# [Upgrade Assistant] Fix test errors

## Summary

This PR addresses two errors that are shown when the tests are executed:

#### Error in reindex_deprecation_flyout [#114691](https://github.com/elastic/kibana/issues/114691)
In which basically were using a paragraph to show the title of the reindexing steps, but step 3 (ReindexingStep) its a somehow more complex title that contains a button and some flex boxes to align everything. So in order to address that, I've changed the paragraph to a simple div. 

#### Error in cluster_upgrade [#114964](https://github.com/elastic/kibana/issues/114964)
This one is basically because when we [capture that](https://github.com/elastic/kibana/blob/7.x/x-pack/plugins/upgrade_assistant/public/application/lib/api.ts#L37) the request returns an error we let the listeners of these changes know about it. In our case in UA when we detect that an error is 426 (upgrade in progress or already upgraded) we unmount UA and show an interstitial. The problem here is that the [use_request hook](https://github.com/elastic/kibana/blob/b7bec6fdb8dd3a96ddeb18a6964230d251d15fdc/src/plugins/es_ui_shared/public/request/use_request.ts#L98) sets the error first and after that it changes the loading state to false. And since the component was unmounted after the error is set, changing the loading state fails. In order to address that, I've added a check before changing the loading state to check if the component is mounted.


#### How to test

1. Checkout this branch
2. Run all the jest tests and make sure that no errors are being outputted to console
